provider/cluster_rosa_classic_resource.go

```diff
@@ -109,21 +111,31 @@ func (t *ClusterRosaClassicResourceType) GetSchema(ctx context.Context) (result
 				Optional:    true,
 				Computed:    true,
 				PlanModifiers: []tfsdk.AttributePlanModifier{
+					tfsdk.UseStateForUnknown(),
 					ValueCannotBeChangedModifier(),
 				},
 			},
 			"name": {
 				Description: "Name of the cluster. Cannot exceed 15 characters in length.",
 				Type:        types.StringType,
-				Required:    true,
+				Optional:    true,
```

Comment:
Cluster name is a required attribute (in rosa cli and the OCM API)
There is a reason you changed it to optional?

Reply:
Required fields must be supplied in the config file and cannot be pulled in via Read(). Leaving this as required (and any of the others as well) would force them to be provided to the import command, in addition to the cluster_id.
To keep the import concise, I only require the cluster_id, and allow read() to populate all the other fields--- that only works if `computed` is true.

The validation now ensures that the formerly `required` fields are defined, it just doesn't differentiate how they get their value (from the user config or from querying the cluster via read()). Those fields will still need to be provided on cluster creation, and that will be enforced.

Reply:
How other providers do this?
I couldn't find any other good examples, as this provider using very old version

Reply:
Regarding docs, yes, autogeneration will move these fields to the optional category. I'm not sure that's actually a bad thing though. I believe there is a mismatch between what terraform means by "required" and what we are trying to convey...

Terraform => "**User** must provide a value"
Us => "This field must have a value"

The distinction really only applies when the resource is imported.

----

It may be possible to change these fields back to required and force the user to put them into the config file prior to import, but that is both redundant and doesn't follow hashi's described flow. However, I'm happy to give it a try if that's how you want it to work.

Reply:
That's why we had the "get" call to ocm in order to supply the required params

Reply:
I'm not understanding...
The only thing we need in order to do the "refresh" is the cluster id, so that's all I ask for at the import step.
Once import calls Read() to do the refresh, all the other fields get populated into the state file (including name, etc.).

The roadblock I hit was that even though `name` was in the state, I got an error that it was a "required" parameter when I ran TF apply the next time (e.g., to scale the cluster).

Reply:
After finding a different explanation/walkthrough of the import process, this is not needed. I have removed it.